src/pal/pal_posix.h

```diff
@@ -117,9 +121,14 @@ namespace snmalloc
      * Bitmap of PalFeatures flags indicating the optional features that this
      * PAL supports.
      *
-     * POSIX systems are assumed to support lazy commit.
+     * POSIX systems are assumed to support lazy commit. The build system checks
+     * getentropy is available, only then this PAL supports Entropy.
      */
-    static constexpr uint64_t pal_features = LazyCommit | Entropy;
+    static constexpr uint64_t pal_features = LazyCommit
+#ifdef LIBC_HAS_GETENTROPY
+      | Entropy
+#endif
+      ;
```

Comment:
This could be simplified by defining the `SNMALLOC_PLATFORM_HAS_GETENTROPY` to either `0` or `Entropy` in the cmake.  Then this entire line becomes:

```suggestion
    static constexpr uint64_t pal_features = LazyCommit | SNMALLOC_PLATFORM_HAS_GETENTROPY;
```

That said, it makes me a little bit nervous to have a global configuration affecting code that's shared among platform configurations.  We support having multiple PAL instantiations in a single program and I can see some confusions arising from this, though those corner cases are not very likely and shouldn't block this.